### DISCUSSION TITLE
New user issues

### DISCUSSION THREAD

**Discussion by cfjedimaster** [beginner-corp/begin-community#94] [Support]
I'm a new user and I wanted to share some pain points I had with getting started.

1) I started with the Quick Start (https://docs.begin.com/en/guides/quickstart). When creating a new project, I don't see similar options as the guide, I do see personal website, but I was looking for "RESTful API" as I wanted to build a simple function. Not the end of the world, but it was confusing.

2) I selected the Hello World template. It created the project just fine. The UI that talks about my first build has a "Edit on GitHub" link that goes to a 404: https://github.com/cfjedimaster/begintest2/edit/main/src/http/get-index/index.js#L21

Notice it's going to src/http/etc.  That leads me to point 3.

3) In the QS, it talks about project structure, a top level public and src folder. Maybe I picked the wrong project, but I have none of that. I've got an index.js file and that's it (I mean outside of the readme, package.json, etc). It looks nothing like the expected project structure. My plan had to been to add my own function based on what's documented under Project Structure, but without a public/src folder I'm not sure it's going to work.

So did I just pick a bad project to start with?

**Comment by cfjedimaster** [comment]
Continuing on - I've now got a project (https://github.com/cfjedimaster/begintest3). It works locally, cool. I want to add a HTTP route like the QS suggests. So I made a `src/http/get-foo` folder. I made an index.js file. In theory, this means I can do a GET to /foo. But - I don't know what these functions look like, ie, what's the basic form of a function in Begin.

So I go to the docs and end up here, https://docs.begin.com/en/http-functions/provisioning/. This seems to imply I _cant_ just make a folder and files, that I have to do something? But it tells me: 

"Open the package.json file in the root of your project."

But the code it suggests:

```
{
  "arc": {
    "app": "myapp",
    "http": [
      [ "get": "/api" ]
    ]
  }
}
```

Where does that go in package.json? Is it at the root? If so, I guess I need to add `arc`? But I'd assume a default Begin app would have at least an empty `arc` part, right?

**Reply by cfjedimaster** [reply]
Ugh I'm sorry - still kinda new to GH DIscussions. I meant the above to be a comment, not an answer.

**Comment by brianleroux** [comment]
- reported `.1` and `.2` and we'll get on fixing those. thx!!
- `.3` you are getting confused by our docs because they suggest a particular structure that is not present. Sorry about the confusion. For years we got critical feedback about the structure of projects so we made EVERYTHING configurable…but its opt-in. Some examples are now using that capability and some are not. [You can read about custom source paths here.](https://arc.codes/docs/en/guides/developer-experience/custom-source-paths)

Related to this is the manifest file. Begin deploys Architect projects which can use any of `app.arc`, `.arc`, `arc.json`, `package.json`, `arc.yaml` or `arc.toml`. If you add routes to it they will work but you also need to add the code. Usually running `arc init` will generate that code for you.

**Reply by cfjedimaster** [reply]
Ok... so maybe my issue was my path in reading? I started with the Quick Start. I then went to the Docs on functions. It feels like maybe the path from QS to... whatever is next is a bit rough? If that makes sense. I'd love to see the QS demonstrate an app with one simple HTML page and one simple serverless function.

**Reply by brianleroux** [reply]
yeah we have a big docs rewrite on the backlog…its def clunky af right now; thanks for sticking w us @cfjedimaster !
